src/js/datepicker.js

```
@@ -110,9 +110,13 @@
 
         this.opts = $.extend(true, {}, defaults, options, this.$el.data());
 
-        if ($body == undefined) {
-            $body = $('body');
-        }
+        if (this.opts.renderTo) {
+            $body = $(this.opts.renderTo);
+        } else {
+            if ($body == undefined) {
```

If the selector fails to match we shouldn't try to hide the error and guess what would be a 'sensible' fallback. This is bound to cause problems. We should signal a warning instead (maybe throw).

---

This is still hasn't been addressed

---

@dennisbot The link 404's now, but there is no code to handle when `If the selector fails to match` as requested above. Sorry I missed your reply

---

I think Javier was referring to this:

```javascript
$domContainer = $(this.opts.renderTo);
```

Which is unsafe because if `this.opts.renderTo` didn't have a valid element as a container, it would crash. Can we update it to something like this?

```javascript
if (this.opts.renderTo) {
  throw "Invalid container element provided";
}
$domContainer = $(this.opts.renderTo);
```